### DISCUSSION TITLE
Pint 0.23 focus is on performance

### DISCUSSION THREAD

**Discussion by hgrecco** [hgrecco/pint#1820] [Announcements]
Pint 0.23 will focus on performance on common tasks. I have prototyped some ideas, which you can find in the develop branch. Briefly, the main thing is refactoring certain internal parts of the code to make it easier to put a cache on costly operations. Please, let us know here what are the largest bottlenecks that you are seeing.

**Comment by matroscoe** [comment]
our biggest bottle neck is the constant conversion from one unit to another. It would be nice if we could set the units/prefix for a dimension on application start and then pint was there to ensure the units work out as expected but only convert them when required. This would allow us to work on ensuring our code still has the units we want but gives an easy path to optimization by fixing methods that are constantly requiring the unit conversion to ensure it is no longer required.  

@hgrecco would be very happy to continue to test out and post flame charts of what we had seen earlier if you give me a shout when you have a commit on `develop` you think would be good to test out.

**Comment by MichaelTiemannOSC** [comment]
@matroscoe just re-upping @hgrecco 's suggestion about putting some code into pint/testsuite/benchmarks.  I made some benchmark code for Pint-Pandas that showed both some surprising good results as well as some expected areas for improvement.

**Reply by matroscoe** [reply]
I wish I was able to but due to time constraints we had to move on from using Pint due to the performance impacts it was having on our software. So unfortunately I am no longer able to spend work time on this due to us moving onto other unit management solutions.
